src/DotNetty.Codecs/Base64/Base64.cs

```diff
@@ -0,0 +1,365 @@
+            fixed (byte* alphabet = dialect.alphabet)
+            {
+                if (src.IoBufferCount == 1 && dest.IoBufferCount == 1)
+                {
+                    destLength = EncodeUsingPointer(alphabet, src, dest, offset, length, breakLines);
+                }
+                else
+                {
+                    destLength = EncodeUsingGetSet(alphabet, src, dest, offset, length, breakLines);
+                }
+            }
+            return dest.SetIndex(0, destLength);
+        }
+
+
+        public static IByteBuffer Decode(IByteBuffer src)
+        {
+            return Decode(src, Base64Dialect.STANDARD);
+        }
+
+        public static IByteBuffer Decode(IByteBuffer src, Base64Dialect dialect)
+        {
+            return Decode(src, src.ReaderIndex, src.ReadableBytes, dialect);
+        }
+
+        public static IByteBuffer Decode(IByteBuffer src, int offset, int length, Base64Dialect dialect)
+        {
+            return Decode(src, offset, length, dialect, src.Allocator);
+        }
+
+
+        private unsafe static int DecodeUsingPointer(IByteBuffer src, IByteBuffer dest, sbyte* decodabet, int offset, int length)
+        {
+            int charCount = 0;
+            fixed (byte* srcArray = src.Array, destArray = dest.Array)
+            {
+                byte* b4 = stackalloc byte[4];
+                int b4Count = 0;
+                int i = 0;
+
+                for (i = offset; i < offset + length; ++i)
+                {
+                    sbyte value = (sbyte)(srcArray[i] & 0x7F);
+                    if (decodabet[value] < WHITE_SPACE_ENC)
+                    {
+                        throw new ArgumentException(String.Format("bad Base64 input character at {0}:{1}",
+                            i, value));
+                    }
+                    if (decodabet[value] >= EQUALS_SIGN_ENC)
+                    {
+                        b4[b4Count++] = (byte)value;
+                        if (b4Count <= 3) continue;
+
+                        if (b4[2] == EQUALS_SIGN)
+                        {
+                            int output = (decodabet[b4[0]] & 0xFF) << 18 |
+                                         (decodabet[b4[1]] & 0xFF) << 12;
+                            destArray[charCount++] = (byte)((uint)output >> 16);
+                        }
+                        else if (b4[3] == EQUALS_SIGN)
+                        {
+                            int output = (decodabet[b4[0]] & 0xFF) << 18 |
+                                         (decodabet[b4[1]] & 0xFF) << 12 |
+                                         (decodabet[b4[2]] & 0xFF) << 6;
+                            destArray[charCount++] = (byte)((uint)output >> 16);
+                            destArray[charCount++] = (byte)((uint)output >> 8);
+                        }
+                        else
+                        {
+                            int output = (decodabet[b4[0]] & 0xFF) << 18 |
```

Comment:
`>>` must use uint, so the lines below use uint to do the rigth shift.
decodabet is sbyte, so when i change output as uint, the code is similar like this:
```
uint output = (uint)(decodabet[b4[0]] & 0xFF) << 18 |
              (uint)(decodabet[b4[1]] & 0xFF) << 12 |
              (uint)(decodabet[b4[2]] & 0xFF) << 6;
destArray[charCount++] = (byte)(output >> 16);
destArray[charCount++] = (byte)(output >> 8);
```